clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/Applications.java

```diff
@@ -88,6 +117,14 @@ public CloudFoundryServerGroup findById(String guid) {
       .collect(toList());
   }
 
+  @Nullable
+  public CloudFoundryServerGroup findServerGroupByNameAndSpaceId(String name, String spaceId) {
```

Comment:
`findServerGroupId` hits API, but `findById` uses the cache. Is there potential drift between them? If we are going to limit what `findServerGroupByNameAndSpaceId` can find to those things in the cache, can we instead just iterate over the cache looking for this name/space combination? And a view on this cache if you think this is called frequently enough to optimize?